app/org/sagebionetworks/bridge/services/email/EmailSignInEmailProvider.java

```diff
@@ -0,0 +1,59 @@
+package org.sagebionetworks.bridge.services.email;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeBodyPart;
+
+import org.sagebionetworks.bridge.BridgeUtils;
+import org.sagebionetworks.bridge.config.BridgeConfigFactory;
+import org.sagebionetworks.bridge.models.studies.Study;
+
+import com.google.common.collect.Maps;
+
+public class EmailSignInEmailProvider implements MimeTypeEmailProvider {
+
+    private final Study study;
+    private final String recipientEmail;
+    private final String subject;
+    private final String body;
+    private final String token;
+    
+    public EmailSignInEmailProvider(Study study, String recipientEmail, String subject, String body, String token) {
+        this.study = study;
+        this.recipientEmail = recipientEmail;
+        this.subject = subject;
+        this.body = body;
+        this.token = token;
+    }
+    
+    @Override
+    public MimeTypeEmail getMimeTypeEmail() throws MessagingException {
+        Map<String,String> map = Maps.newHashMap();
+        map.put("studyName", study.getName());
+        map.put("supportEmail", study.getSupportEmail());
+        map.put("technicalEmail", study.getTechnicalEmail());
+        map.put("sponsorName", study.getSponsorName());
+        map.put("token", token);
+        map.put("host", BridgeConfigFactory.getConfig().getHostnameWithPostfix("webservices"));
```

Comment:
Not necessary for this change, but I've noticed that every call to BridgeUtils.resolveTemplate() has a similar but different map. It might be worthwhile filing a JIRA to refactor BridgeUtils.resolveTemplate() to take in a study and fill in a basic common set of template vars, so we don't have to re-define them with every call.